cluster-state-service/handler/event/kinesis_consumer_test.go

```diff
@@ -0,0 +1,139 @@
+	kinesisClient            *mocks.MockKinesisAPI
+	processor            *mocks.MockProcessor
+	getShardIteratorInput     *kinesis.GetShardIteratorInput
+	getShardIteratorOutput    *kinesis.GetShardIteratorOutput
+	getRecordsInput  *kinesis.GetRecordsInput
+	getRecordsOutput *kinesis.GetRecordsOutput
+	record1 *kinesis.Record
+	record2 *kinesis.Record
+}
+
+func NewConsumerMockKinesisContext(t *testing.T) *consumerMockKinesisContext {
+	context := consumerMockKinesisContext{}
+	context.mockCtrl = gomock.NewController(t)
+	context.kinesisClient = mocks.NewMockKinesisAPI(context.mockCtrl)
+	context.processor = mocks.NewMockProcessor(context.mockCtrl)
+
+	context.record1 = &kinesis.Record{
+		Data: []byte(kinesisMessageBody1),
+	}
+
+	context.record2 = &kinesis.Record{
+		Data: []byte(kinesisMessageBody2),
+	}
+
+	context.getShardIteratorInput = &kinesis.GetShardIteratorInput{
+		ShardId: aws.String("shardId-000000000000"),
+		ShardIteratorType: aws.String("TRIM_HORIZON"),
+		StreamName: aws.String(streamName),
+	}
+
+	context.getShardIteratorOutput = &kinesis.GetShardIteratorOutput{
+		ShardIterator: aws.String("shardId-123"),
+	}
+
+	context.getRecordsInput = &kinesis.GetRecordsInput{
+		Limit: aws.Int64(100),
+		ShardIterator: aws.String("shardId-123"),
+	}
+
+	context.getRecordsOutput = &kinesis.GetRecordsOutput{
+		Records: []*kinesis.Record{context.record1, context.record2},
+	}
+
+	return &context
+}
+
+func TestNewConsumerNilKinesis(t *testing.T) {
+	context := NewConsumerMockKinesisContext(t)
+	defer context.mockCtrl.Finish()
+
+	_, err := NewKinesisConsumer(nil, context.processor, streamName)
+	if err == nil {
+		t.Error("Expected an error when kinesis is nil")
+	}
+}
+
+func TestNewConsumerKinesisNilProcessor(t *testing.T) {
+	context := NewConsumerMockKinesisContext(t)
+	defer context.mockCtrl.Finish()
+
+	_, err := NewKinesisConsumer(context.kinesisClient, nil, streamName)
+	if err == nil {
+		t.Error("Expected an error when processor is nil")
+	}
+}
+
+func TestNewConsumerKinesisEmptyQueueName(t *testing.T) {
+	context := NewConsumerMockKinesisContext(t)
+	defer context.mockCtrl.Finish()
+
+	_, err := NewKinesisConsumer(context.kinesisClient, context.processor, "")
+	if err == nil {
+		t.Error("Expected an error when stream name is empty")
+	}
+}
+
+func TestPollForKinesisEventsReceiveTwoMessages(t *testing.T) {
+	mockContext := NewConsumerMockKinesisContext(t)
+	defer mockContext.mockCtrl.Finish()
+
+	mockContext.kinesisClient.EXPECT().GetShardIterator(gomock.Eq(mockContext.getShardIteratorInput)).Return(mockContext.getShardIteratorOutput, nil)
+
+	c, err := NewKinesisConsumer(mockContext.kinesisClient, mockContext.processor, streamName)
+
+	if err != nil {
+		t.Errorf("Unexpected error when calling NewConsumer: %+v", err)
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	pollCount := 0
+
+	mockContext.kinesisClient.EXPECT().GetRecords(mockContext.getRecordsInput).Return(mockContext.getRecordsOutput, nil)
```

Comment:
Can we also tests to verify error handling when each of these mocks returns errors?